### PR TITLE
Use lint-staged

### PR DESCRIPTION
**Pull request by shvaikalesh**
Cool, `lint-staged` does not break GitHub Desktop client.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,18 +55,27 @@
...
+    "lint-staged": "lint-staged",
     "prepublish": "gulp build",
     "build": "gulp build",
     "lint": "gulp lint",
     "test": "gulp unit && nightwatch"
+  },
+  "pre-commit": "lint-staged",
+  "lint-staged": {
+    "lib/**/*.js": "eslint",
+    "*": "lintspaces -e .editorconfig -i js-comments"
```
Comment (shvaikalesh) [inline package.json]: We will simplify globs later.

**Comment by pepelsbey** [comment]
Look @okonet :)

**Comment by okonet** [comment]
👏

**Comment by okonet** [comment]
Even better with:

```json
{
    "*.js": ["eslint --fix", "git add"]
}
```

**Comment by shvaikalesh** [comment]
Updated the PR. @okonet thanks for the tool 👍

**Comment by pepelsbey** [comment]
LGTM

**Pull request merged**